src/phases/query/CollScanComplexPredicate.yml

```diff
@@ -0,0 +1,521 @@
+          {$and: [{h: {$in: [123, 321]}},
+                  {$or: [{$and: [{h: {$gt: 23}}, {h: {$lt: 1023}}, {f: 99}]},
+                        {$and: [{h: {$gt: 923}}, {h: {$lt: 2023}}, {b: 99}]},
+                        {$and: [{h: {$gt: 1923}}, {h: {$lt: 3023}}, {a: 99}]}]},
+                  {$and: [{$and: [{f: {$gt: 23}}, {f: {$lt: 1023}}, {p: 99}]},
+                        {$and: [{f: {$gt: 923}}, {f: {$lt: 2023}}, {q: 99}]},
+                        {$and: [{f: {$gt: 1923}}, {f: {$lt: 3023}}, {r: 99}]}]},
+                  {$or: [{$and: [{d: {$gt: 23}}, {d: {$lt: 1023}}, {d: 'fhr'}]},
+                        {$and: [{d: {$gt: 923}}, {d: {$lt: 2023}}, {e: 9}]},
+                        {$and: [{d: {$gt: 1923}}, {d: {$lt: 3023}}, {f: 11}]}]}
+                  ]}
+        ]}
+    ]}
+
+  MixedQuery11LevelsDeep: &MixedQuery11LevelsDeep
+    {$and: [
+      {b: 0},
+      {$or: [{a: {$in: [17, 15, 1, 8, 2]}}, {c: {$in: ['ab', 'dh', 'ef']}}]},
+      {$and: [
+          {a: {$gt: 42}}, {c: {$lte: 'ef'}},
+          {$or: [
+              {a: 2}, {a: 1},
+              {$and: [
+                  {c: {$in: ['34', '35', '42', '23', '17', '19', 'abc']}}, {a: {$in: [34, 35, 36, 37, 67]}}
+              ]},
+              {$and: [
+                  {a: {$in: [34, 35, 36, 37, 67]}},
+                  {$or: [{$and: [{d: {$gt: '123hab'}}, {d: {$in: ['aaa', '13217', 'zzz13']}}]},
+                    {$and: [{a: {$in: [34, 35, 46, 57]}}, {b: 1}, {c: '42123'}]},
+                    {$or: [{f: {$in: [1, 3, 7, 8]}}, {e: 'vbg321'}, {d: 'hbr321'}]},
+                    {d: {$gt: 'habasdf'}}, {d: {$lte: 'hia123'}},
+                    {$and: [
+                      {d: {$in: ["7", "5", "6", "9", "2"]}},
+                      {d: {$in: ['123123', 109, 110]}},
+                      {$and: [{$or: [
+                          {$and: [{a: {$lt: 11}}, {c: {$gt: 12}}, {f: {$in: ["1", "2", 1, 2]}}, {d: null}]},
+                          {h: {$gt: 0}},
+                          {h: {$lt: 100}},
+                          {f: 99},
+                          {$or: [
+                            {f: {$lt: 1}}, {h: {$eq: 12}}, {f: {$in: ["1", "2", 1, 2]}}, {a: {$lte: null}}, 
+                            {$and: [
+                              {a: {$gt: -99}}, {b: {$lt: 12}}, {c: {$in: [null, 1, -1, "123"]}}, {b: {$gt: null}}, 
+                              {$or: [
+                                {$and: [{f: {$gt: 0}}, {f: {$lt: 100}}, {p: 99}]},
+                                {$or: [{f: {$gt: 90}}, {f: {$lt: 200}}, {q: 99}]},
+                                {$and: [{f: {$gt: 190}}, {f: {$lt: 300}}, {r: 99}]}
+                              ]}
+                            ]}
+                          ]}
+                        ]},
+                        {$and: [{h: {$gt: 90}}, {h: {$lt: 200}}, {b: 99}]},
+                        {$or: [
+                          {$and: [{a: {$lt: 11}}, {c: {$gt: 12}}, {f: {$in: ["1", "2", 1, 2]}}, {d: null}]},
```

Comment:
I think Genny might still have a bug where the literal "1" is turned into the number 1 directly. I'd suggest using a different string. Speaking of strings- can we make one of the fields be a string field? Could have one "shortString" field and one "longString" field, since the collscan benchmarks doc mentioned that we don't have enough coverage on the string type.